### DISCUSSION TITLE
how can response.button give not just the index of the selection made, but the value of the button itsef?

### DISCUSSION THREAD

**Discussion by auraangelica** [jspsych/jsPsych#1501] [Q&A] [answered]
Hi all: I'm brand new to this so apologies for ugly formatting. 

I am making an experiment in which subjects press a button for the word they think is the prettiest out of three choices. Right now, when I click, the data outputted by `"button_pressed": response.button` is the index of the button that was pressed (i.e., 0 1 or 2 depending on if they chose the first second or third word).  Rather than (only) getting the index of the button that was pressed, I am interested in my output also including the word (string) itself. 

Basically, I'd like to output a "word_selected" equivalent to "button_pressed". Any ideas on how I can do that? 

(I'm going off of _jspscyh-html-button-response.js_)

**Comment by jodeleeuw** [comment]
Hi @auraangelica,

We should make this an official feature of the plugins. I actually thought we already did do this, but after looking more carefully I just imagined that we had implemented this.

Once version 6.3 comes out, within the next week or so, it'll be easier to do this because of the new feature that allows saving any trial parameter in the data. Once that feature is launched, you can save the `choices` array and use that information to get the value of the button like this:

```js
var button_trial = {
	type: 'html-button-respose',
	stimulus: '<p>Click a button, any button!</p>',
	choices: ['A', 'B', 'C', 'D'],
	save_trial_parameters: {
		choices: true
	},
	on_finish: function(data){
		data.response_label = data.choices[data.response]
	}
}
```

Note that I'm using `data.response` instead of `data.button_pressed` because we are standardizing all of the response data names to `response`.

**Reply by auraangelica** [reply]
Thanks @jodeleeuw ! 

In that case, does this mean I should just wait until 6.3? Thanks :)

**Comment by auraangelica** [comment, marked as answer]
In case anyone's wondering, adding this to the `jspsych-html-button-response.js` plugins file, under `var trial_data` worked for 6.2! 

`        "word_chosen": trial.choices[response.button], 
` 

(Full code block) 

> // gather the data to store for the trial
      var trial_data = {
        "rt": response.rt,
        "stimulus": trial.stimulus,
        "button_pressed": response.button,
        "word_chosen": trial.choices[response.button], 
      };
